### PR TITLE
Update ActiveRecord::QueryMethods#in_order_of example [ci-skip]

### PR DESCRIPTION
**Pull request by sampatbadhe**
### Detail

Update `ActiveRecord::QueryMethods#in_order_of` example
Replace `ORDER BY FIELD` example with `ORDER BY CASE`
`ORDER BY FIELD` was MYSQL specific, which was replaced with default `ORDER BY CASE` in https://github.com/rails/rails/pull/45670

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] CI is passing.

### CODE REVIEW

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 1]
```diff
@@ -512,13 +512,12 @@ def order!(*args) # :nodoc:
...
 
-    # Allows to specify an order by a specific set of values. Depending on your
-    # adapter this will either use a CASE statement or a built-in function.
+    # Allows to specify an order by a specific set of values.
     #
     #   User.in_order_of(:id, [1, 5, 3])
     #   # SELECT "users".* FROM "users"
-    #   #   ORDER BY FIELD("users"."id", 1, 5, 3)
     #   #   WHERE "users"."id" IN (1, 5, 3)
+    #   #   ORDER BY CASE WHEN "users"."id" = 1 THEN 1 WHEN "users"."id" = 5 THEN 2 WHEN "users"."id" = 3 THEN 3 END ASC
```
Comment (jonathanhefner) [inline activerecord/lib/active_record/relation/query_methods.rb:520]: This ends up being a bit long when rendered:

apidocs-in_order_of

What do you think about breaking it up?

```suggestion
    #   #   ORDER BY CASE
    #   #     WHEN "users"."id" = 1 THEN 1
    #   #     WHEN "users"."id" = 5 THEN 2
    #   #     WHEN "users"."id" = 3 THEN 3
    #   #   END ASC
```

**Review by jonathanhefner — COMMENTED** [review]
Good catch! :+1:

File: activerecord/lib/active_record/relation/query_methods.rb (hunk 1 above)
Comment (sampatbadhe) [reply activerecord/lib/active_record/relation/query_methods.rb:520]: agree @jonathanhefner I'll update.

**Review by jonathanhefner — APPROVED** [review]

**Comment by jonathanhefner** [comment]
Thank you, @sampatbadhe! :wave:

**Pull request merged**